### PR TITLE
Add torch.multiprocessing.create_processes

### PR DESCRIPTION
**Pull request by ailzhang**

### CODE REVIEW

**Review by soumith — CHANGES_REQUESTED** [review]
the function itself is `spawn`, so I don't get why this is added

**Comment by thomasbrandon** [comment]
Might still be WIP but related to [this `torch_xla` issue](https://github.com/pytorch/xla/issues/1217).In notebooks

**Comment by ailzhang** [comment]
@soumith How about adding a more general API like `create_processes` as shown in the latest commit? 
I also renamed `SpawnContext` to `ProcessContext` since it's not really specific to `spawn`. 
This change will allow environments like jupyter notebook(like Colab) to use torch.multiprocessing as well.

**Review by soumith — COMMENTED** [review]
I want to give some context about `torch.multiprocessing`. We mirror upstream Python `multiprocessing`'s API, and this is deliberate. The thing we do additionally is that we route the pickling through our custom pickler.

So, if we are changing something like SpawnContext -> ProcessContext, and changing any existing function signatures, @ailzhang please confirm that we are not changing the API signatures of functions that are mirrored from upstream Python `multiprocessing`

File: torch/multiprocessing/spawn.py [hunk 1]
```diff
@@ -118,7 +118,39 @@ def join(self, timeout=None):
...
 
-def spawn(fn, args=(), nprocs=1, join=True, daemon=False):
+class SpawnContext(ProcessContext):
+    def __init__(self, processes, error_queues):
+        warnings.warn('SpawnContext is renamed to ProcessContext since 1.4 release.')
+        super(SpawnContext, self).__init__(self, processes, error_queues)
+    pass
+
+
+def start_processes(fn, args=(), nprocs=1, join=True, daemon=False, start_method='spawn'):
```
Comment (SsnL) [inline torch/multiprocessing/spawn.py]: Is this meant to be public? If so, maybe add doc entries and move the docstring to here?

**Comment by ailzhang** [comment]
@soumith Hmmm python multiprocessing doesn't have a `.spawn()` function/method. It 's only available as an option in keyword arg `start_method`. Do you know which Python API we are mirroring here?

**Comment by ailzhang** [comment]
@SsnL @apaszke Could you help confirm whether it's okay to rename `SpawnContext` and the changes in this PR? Thank you!

**Comment by SsnL** [comment]
@ailzhang Yeah, I think it is fine.

**Comment by SsnL** [comment]
I think spawncontext was added to be used as a helper for distributed packages. It is not in the python stdlib.

**Comment by SsnL** [comment]
If start_process is not supposed to be part of a public API, I think this PR is good to go.

**Comment by ailzhang** [comment]
> If start_process is not supposed to be part of a public API, I think this PR is good to go.

Yep I'm thinking about keep it not documented for now, and XLA will use it mainly for colab training for 8TPU cores. If it gets popular we can refine and add documentation to make it public?

**Review by ssnl — APPROVED** [review]

**Review by ssnl — APPROVED** [review]
LGTM!

**Pull request closed without merging**